ipywidgets/static/widgets/js/manager.js

```diff
@@ -280,11 +280,26 @@ define([
         /**
          * Handle when a comm is opened.
          */
-        return this.create_model({
-            model_name: msg.content.data.model_name,
-            model_module: msg.content.data.model_module,
-            comm: comm,
-        }).catch(utils.reject("Couldn't create a model.", true));
+        // For < 4.x
+        var model;
+        if (msg.content.data.model_name) {
```

Comment:
About different versions of the back-end in the front end: You definitely don't need the code to sit around for all releases. I think deprecating (printing a message to the console) in the upcoming release and then removing the thing in the next release is good. At least, even a changelog / NEWS entry will be very helpful.